### ISSUE TITLE
Changing colorscheme causes jumbled text in the buffer

### ISSUE THREAD

**Issue by Kiran-Raj-Dev** [neovim/neovim#19573]
### Neovim version (nvim -v)

NVIM v0.8.0-dev-738-g02efdb4d5

### Vim (not Nvim) behaves the same?

no, vim 8.2 (included 

### Operating system/version

macOS 12.4

### Terminal name/version

kitty 0.25.2

### $TERM environment variable

xterm-kitty

### Installation

built from source

### How to reproduce the issue

```zsh
# note replace the ^[ with the escape character
echo "i🏄‍♂️^[jyy100p:set mouse=a number" | nvim --clean -s - 
```
Now scroll around a bit
then do `:colorscheme darkblue` and then scroll a bit as well.

### Expected behavior

Note: This is using the default colorscheme

https://user-images.githubusercontent.com/90276965/181769743-e4b378a1-bcca-4ca2-8450-a48bc67c7eaa.mov

### Actual behavior

Note: This is using the darkblue colorscheme (but in my testing any colorscheme other than the default one causes this problem)

https://user-images.githubusercontent.com/90276965/181770024-0985c584-cffa-40e0-b2a6-a003af10a98e.mov

**Comment by zeertzjq** [comment]
Duplicate of #7151

**Comment by zeertzjq** [comment]
Note: after #19686 this bug is no longer reproducible.

**Comment by Kiran-Raj-Dev** [comment]
Yup, I can confirm it is no longer reproducible. Thanks for the fix!
